docs/source/install/upgrades.rst

```diff
@@ -1,33 +1,101 @@
 migration scripts for picking up data model changes.
 
-For your convenience, the list of migrations to run when upgrading to a version is listed in docs.
-See :ref:`migration scripts to run<migration-scripts-to-run>` for details. If
-you skipped a version and are upgrading to a newer version, please make sure you *run migration
+The list of migrations to run when upgrading to a version is listed in the
+:ref:`section below<migration-scripts-to-run>`.
+
+If you skipped a version and are upgrading to a newer version, please make sure you *run migration
 scripts* for skipped versions as well.
 
-.. _all-in-one-to-packages:
 
-I am using all-in-one (AIO) installer or st2_deploy script based installer
---------------------------------------------------------------------------
+Upgrade Procedure
+-----------------
+
+:ref`Migration scripts<migration-scripts-to-run>` most often need to be run when upgrading to |st2|
+:versions that include data model changes. The typical **upgrade** procedure is
+
+1. Stop ``st2*`` services on the box.
+
+.. sourcecode:: bash
+
+   sudo st2ctl stop
+
+2. Run the migration script (if any). See section below for |st2|
+   version-specific migration scripts.
+
+3. Upgrade |st2| packages (``st2``, ``st2web``, `st2chatops``, ``st2mistral``
+   and ``bwc-enterprise`` using distro specific tools.
+
+  Ubuntu:
+
+  .. sourcecode:: bash
+
+     sudo apt-get install --only-upgrade $PKG_NAME
+
+  RHEL / CentOS:
+
+  .. sourcecode:: bash
+
+     sudo yum update $PKG_NAME
+
+4. Upgrade Mistral database.
+
+.. sourcecode:: bash
 
-The safe and reliable way to transition |st2| to newer versions is to provision a
-new |st2| instance, and roll over the content. Thanks to the "Infrastructure as code"
-approach, all |st2| content and artifacts are simple files, and should be kept under source
-control.
+  # Stop related services
+  service mistral-api stop
+  service mistral stop
+
+  # Upgrade database
+  /opt/stackstorm/mistral/bin/mistral-db-manage --config-file /etc/mistral/mistral.conf upgrade head
+  /opt/stackstorm/mistral/bin/mistral-db-manage --config-file /etc/mistral/mistral.conf populate
+
+  # Restart related services
+  service mistral start
+  service mistral-api start
+
+5. Start |st2| services.
+
+.. sourcecode:: bash
+
+   sudo st2ctl start
+
+.. _migration-scripts-to-run:
+
+Version-specific migration scripts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We document :ref:`upgrade notes<upgrade_notes>` for the various versions. The upgrade
+notes section gives an idea of what major changes happened with each release. You may also want
+to take a look at detailed :doc:`/changelog` for each version.
+Following sections call out the migration scripts that need to be run before upgrading to the
+respective version
+
+v1.5
+'''''
+
+* Datastore model migration
+
+::
+
+    /opt/stackstorm/st2/bin/st2-migrate-datastore-to-include-scope-secret.py
+
+Content Roll-Over
```

Comment:
I've been working through some of the heading capitalisations, but there's a lot to fix up... It does matter though. Makes docs more polished.
I was mainly focusing on H1 & H2, since they show up in various TOCs. But need to go through all of them.